File: go/tenniskata/TennisGame1.go
package tenniskata

type tennisGame1 struct {
	m_score1    int
	m_score2    int
	player1Name string
	player2Name string
}

func TennisGame1(player1Name string, player2Name string) TennisGame {
	game := &tennisGame1{
		player1Name: player1Name,
		player2Name: player2Name}

	return game
}

func (game *tennisGame1) WonPoint(playerName string) {
	if playerName == "player1" {
		game.m_score1 += 1
	} else {
		game.m_score2 += 1
	}
}

func (game *tennisGame1) GetScore() string {
	score := ""
	tempScore := 0
	if game.m_score1 == game.m_score2 {
		switch game.m_score1 {
		case 0:
			score = "Love-All"
		case 1:
			score = "Fifteen-All"
		case 2:
			score = "Thirty-All"
		default:
			score = "Deuce"
		}
	} else if game.m_score1 >= 4 || game.m_score2 >= 4 {
		minusResult := game.m_score1 - game.m_score2
		if minusResult == 1 {
			score = "Advantage player1"
		} else if minusResult == -1 {
			score = "Advantage player2"
		} else if minusResult >= 2 {
			score = "Win for player1"
		} else {
			score = "Win for player2"
		}
	} else {
		for i := 1; i < 3; i++ {
			if i == 1 {
				tempScore = game.m_score1
			} else {
				score += "-"
				tempScore = game.m_score2
			}
			switch tempScore {
			case 0:
				score += "Love"
			case 1:
				score += "Fifteen"
			case 2:
				score += "Thirty"
			case 3:
				score += "Forty"
			}
		}
	}
	return score
}

File: vbnet/Tennis/TennisGame1.vb
Public Class TennisGame1
    Implements ITennisGame

    Private m_score1 As Integer = 0
    Private m_score2 As Integer = 0
    Private player1Name As String
    Private player2Name As String

    Public Sub New(ByVal player1Name As String, ByVal player2Name As String)
        Me.player1Name = player1Name
        Me.player2Name = player2Name
    End Sub

    Public Sub WonPoint(ByVal playerName As String) Implements ITennisGame.WonPoint
        If playerName = "player1" Then
            m_score1 += 1
        Else
            m_score2 += 1
        End If
    End Sub

    Public Function GetScore() As String Implements ITennisGame.GetScore
        Dim score As String = ""
        Dim tempScore = 0

        If m_score1 = m_score2 Then

            Select Case m_score1
                Case 0
                    score = "Love-All"
                Case 1
                    score = "Fifteen-All"
                Case 2
                    score = "Thirty-All"
                Case Else
                    score = "Deuce"
            End Select
        ElseIf m_score1 >= 4 OrElse m_score2 >= 4 Then
            Dim minusResult = m_score1 - m_score2

            If minusResult = 1 Then
                score = "Advantage player1"
            ElseIf minusResult = -1 Then
                score = "Advantage player2"
            ElseIf minusResult >= 2 Then
                score = "Win for player1"
            Else
                score = "Win for player2"
            End If
        Else

            For i = 1 To 3 - 1

                If i = 1 Then
                    tempScore = m_score1
                Else
                    score += "-"
                    tempScore = m_score2
                End If

                Select Case tempScore
                    Case 0
                        score += "Love"
                    Case 1
                        score += "Fifteen"
                    Case 2
                        score += "Thirty"
                    Case 3
                        score += "Forty"
                End Select
            Next
        End If

        Return score
    End Function
End Class